frontend/src/models/service.ts

```diff
@@ -0,0 +1,40 @@
+import { createModel } from '@rematch/core'
+import { RootModel } from './rootModel'
+import { emit } from '../services/Controller'
+
+type ServiceParams = { [key: string]: any }
+
+type IServiceState = {
+  port?: number
+  host?: string
+  isValid?: boolean
+  loading?: boolean
+}
+
+const state: IServiceState = {
+  port: undefined,
+  host: undefined,
+  isValid: false,
+  loading: false,
+}
+
+export default createModel<RootModel>()({
+  state,
+  effects: dispatch => ({
+    async checkService(data: IServiceState) {
+      const { set } = dispatch.service
+      try {
+        set({ loading: true })
+        emit('service/check-host-port', data)
+      } catch (error) {
+        set({ isValid: false })
```

Comment:
Let's remove this 'service' model. I think we have a namespace conflict with our existing services. Also we can just use the backend model like we do for freeport.

The emit can move to the component and the isValid could become reachablePort? boolean in the backend model